lib/coll.gi

```diff
@@ -169,14 +169,17 @@ InstallMethod( IsWholeFamily,
 ##
 #M  Size( <C> ) . . . . . . . . . . . . . . . . . . . .  size of a collection
 ##
-InstallImmediateMethod( Size,
-    IsCollection and HasIsFinite and IsAttributeStoringRep, 0,
-    function ( C )
-    if IsFinite( C ) then
-        TryNextMethod();
-    fi;
-    return infinity;
-    end );
+# This method get called for every group that knows it is finite but does
+# not know its size -- e.g. permutation, pc
+# the benefit OTOH is minimal beyond showing off a feature.
+#InstallImmediateMethod( Size,
+#    IsCollection and HasIsFinite and IsAttributeStoringRep, 0,
+#    function ( C )
+#    if IsFinite( C ) then
+#        TryNextMethod();
+#    fi;
+#    return infinity;
+#    end );
```

Comment:
I agree this should not be an immediate method, but perhaps it should be retained as a regular method? Then with the `IsAttributeStoringRep` removed.

Reply:
If we do so, we need to rank this high, so that it overrides any other.

(Also removing this immediate method broke specific tests for this method will need to be adjusted.)